sljit_src/sljitConfigInternal.h

```diff
@@ -971,27 +971,41 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_sw sljit_exec_offset(void *code);
 #endif /* !SLJIT_COMPILE_ASSERT */
 
 #ifndef SLJIT_FALLTHROUGH
-
-#if __cplusplus >= 201703L
-#define SLJIT_FALLTHROUGH [[__fallthrough__]];
-#elif __STDC_VERSION__ >= 202311L \
-	|| (__STDC_VERSION__ > 201710L && defined(__has_c_attribute))
-#if __STDC_VERSION__ >= 202311L || __has_c_attribute(fallthrough) >= 201904L
-#define SLJIT_FALLTHROUGH [[__fallthrough__]];
+#if defined(__cplusplus) && __cplusplus >= 201703L && \
+	defined(__has_cpp_attribute)
+/* Standards-compatible C++ variant. */
+#if __has_cpp_attribute(fallthrough)
+#define SLJIT_FALLTHROUGH [[fallthrough]];
+#endif
+#elif !defined(__cplusplus) && \
+	defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L && \
+	defined(__has_c_attribute)
+/* Standards-compatible C variant. */
+#if __has_c_attribute(fallthrough)
+#define SLJIT_FALLTHROUGH [[fallthrough]];
 #endif
 #elif defined(__clang__)
-#if (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 2)) \
-	&& __cplusplus >= 201103L
+/* Clang-specific variants. */
+#if (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 2)) && \
+	defined(__cplusplus) && __cplusplus >= 201103L && \
+	defined(__has_cpp_attribute)
+#if __has_cpp_attribute(clang::fallthrough)
 #define SLJIT_FALLTHROUGH [[clang::fallthrough]];
-#elif defined(__has_attribute)
-#if ((__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 6)) \
-	&& __has_attribute(fallthrough) && !defined(__cplusplus) \
-	|| (__clang_major__ >= 6 && __cplusplus == 199711L))
-#define SLJIT_FALLTHROUGH __attribute__((__fallthrough__));
-#endif
-#endif /* clang */
-#elif __GNUC__ >= 7 && !defined(__INTEL_COMPILER) && !defined(__NVCC__)
-#define SLJIT_FALLTHROUGH __attribute__((__fallthrough__));
+#endif
+#elif (__clang_major__ > 3 || (__clang_major__ == 3 && __clang_minor__ >= 6)) && \
+	(!defined(__cplusplus) || (__cplusplus == 199711L && __clang_major__ >= 6)) && \
+	defined(__has_attribute)
+#if __has_attribute(fallthrough)
+#define SLJIT_FALLTHROUGH __attribute__((fallthrough));
+#endif
+#endif /* Clang-specific */
+#elif defined(__GNUC__) && __GNUC__ >= 7 && \
+	!defined(__INTEL_COMPILER) && !defined(__NVCC__) && \
+	defined(__has_attribute)
+/* GCC-specific variant. */
+#if __has_attribute(fallthrough)
```

Comment:
this and the previous `__has_attribute` are already warranted by the fact this is `clang`, a GNU compiler and has a version check, do you know of any reason why this might be needed?